tests/test_rpc.py

```diff
@@ -250,9 +250,8 @@ def test_exception_iwth_ssl():
 
 def test_client_timeout():
     with pytest.raises(socket.timeout):
-        with pytest.warns(UserWarning):  # Deprecated
```

Comment:
This `UserWarning` is never been raised, but the older version of pytest will ignore the failure when it't in a `with pytest.raises(...)` context.

So we should just remove it.